### PR TITLE
Changing Debian packaging tool #1325

### PR DESCRIPTION
**Pull request by veb7vmehra**

### CODE REVIEW

**Comment by henrykironde** [comment]
Thanks @veb7vmehra for these changes. Before I do review, have you been able to build a deb package using this setup?

**Comment by veb7vmehra** [comment]
@henrykironde Yes it was working on mine.

**Comment by henrykironde** [comment]
Could you paste some commands and on output on the steps you used to make it easy for me to test?

**Comment by veb7vmehra** [comment]
@henrykironde I guess you can just try by uninstalling retriever and then reinstalling it with debian and then can try running some examples maybe. I have tried like this only.

**Comment by henrykironde** [comment]
I wanted to mean the steps to create the `debian` package.
Could you include some developer documentation.

**Comment by henrykironde** [comment]
To do: 
- Add test cases for json to cvs using the cases we have been looking at.
[Adding test cases](https://github.com/weecology/retriever/blob/master/test/test_retriever.py#L515) is easy since there is an example already.
- Add comments or commands on how to create the debian package so that other mentors test it

**Comment by veb7vmehra** [comment]
You can check debian package via Lintian which you can install from [here](https://github.com/Debian/lintian). Then you can use command `lintian <filename>`  to check it. For more information you can refer [this](https://lintian.debian.org/manual/index.html)

**Comment by veb7vmehra** [comment]
@henrykironde have you checked this on your system cause I found out some more ways with which we can change debian packaging in case this one didn't work up to your expectations

**Comment by henrykironde** [comment]
No I have not yet because I am waiting for the commands to be pasted here

**Comment by veb7vmehra** [comment]
@henrykironde First of all run the following commands to install the build essential package
`sudo apt-get update`
`sudo apt-get install build-essential`
Then you have to install packages listed in build depend fields referred in control.md
Then run the following command in your source directory:
`dpkg-buildpackage -us -uc`
I guess the above part will only work after you make the changes that I specified in the issue #1325 in the dockerfile. If this gives the expected results than use the `debsign` command with your private GPG key.
I guess this will help.

**Comment by henrykironde** [comment]
Okey, I will try to follow the workflow and test these commands tomorrow.

**Comment by veb7vmehra** [comment]
Any updates Sir?

File: debian/control.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# ------------------------------------------------------------------------- #
+# DEBIAN PACKAGE CONTROL FILE                                               #
+#                                                                           #
+# This file is a Debian control file. For more information on the config in #
```
Comment (henrykironde) [inline debian/control.md]: This file needs cleaning up

File: debian/compat.md [hunk 2]
```diff
@@ -0,0 +1 @@
+9
```
Comment (henrykironde) [inline debian/compat.md]: Why do we start from 9

File: debian/ChangeLog.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+python-deb-pkg (0.1.0-1) UNRELEASED; urgency=low
+
+* python-deb-pkg: Initial debian packaging
+
+-- Krzysztof Zuraw <krzysztof.zuraw@gmail.com>  Sat, 3 Jun 2017 12:34:56 +0000
```
Comment (henrykironde) [inline debian/ChangeLog.md]: Clean up

**Comment by veb7vmehra** [comment]
> Let us clean up whole setup. Add [docs for developers here](https://github.com/weecology/retriever/blob/master/docs/developer.rst) so that we get to know why these files are included.
> You can use MAINTAINER Ethan White [ethan@weecology.org](mailto:ethan@weecology.org)
Where should I add the doc, under the heading of Create Release/Linux or should I create a new subheading for that

**Comment by henrykironde** [comment]
We should be adding it here https://github.com/weecology/retriever/blob/master/docs/developer.rst#linux
Remove what does not apply under the old way of creating the deb package

**Comment by ethanwhite** [comment]
Closing since we are dropping support for installation pathways other than PyPI, conda, and source.

**Pull request closed without merging**